python/src/iceberg/expression/literals.py

```diff
@@ -200,78 +205,52 @@ def _(self, type_var):
         return self
 
 
-class IntegerLiteral(Literal[int]):
```

Comment:
lovely, did not realized we can actually ditch this entirely. But looking at the rest of the change altogether, it make more sense as LongLiteral is probably works fine behind the scene and when explicitly convert to IntegerType, we enforce the java like min and max but we are still return LongLiteral in the end but provide a better bound